### PR TITLE
MRC-440: Allow specifying context root id

### PR DESCRIPTION
**Pull request by richfitz**
This PR will allow a root id to be provided when creating a context.  The rationale is explained in the documentation:

> Force a context root id.  This is intended for advanced use only.  By settting the root id, two contexts created with storage in different file locations (\code{path}) will get the same id.  This is required for using a server-hosted database to share a context between different physical machines (or different docker containers).  The id, if provided, must be compatible with \code{ids::random_id()} - i.e., a 32 character hex string.  This option can be left alone in most situations.

This is required to support mrc-439 so that rrq will work nicely without shared disks.

Sorry, this PR lacks a lot of broader context - this is a package that underlies a lot of the cluster tools for specifying and creating a computational environment repeatedly, as well as for serialising the expressions to run remotely

### CODE REVIEW

File: R/task_bulk.R [hunk 1]
```diff
@@ -83,7 +83,7 @@ bulk_prepare_expression_X <- function(X, do_call, use_names) {
       lens <- lengths(X)
       ## Here, support recycling out scalars
       ul <- unique(lens)
-      if (ul == 2L && min(ul) == 1L) {
+      if (length(ul) == 2L && min(ul) == 1L) {
```
Comment (r-ash) [inline R/task_bulk.R:86]: I'm a little unclear on why this change is necessary. Just an opportunistic fix or a change required for the PR?

**Review by r-ash — APPROVED** [review]
Looks good, just one question about something unclear to me.

As a side note I also notice that the website URL for the repo doesn't work. Currently is http://dide-tools.github.io/context

File: R/task_bulk.R (hunk 1 above)
Comment (richfitz) [reply R/task_bulk.R:86]: There was a logical comparison of nonscalars here if there were more than two elements - so this worked sometimes but only by accident. So yes, opportunistic fix, but triggered by updates in R CMD check that now screen for this

Comment (r-ash) [reply R/task_bulk.R]: Great, will merge then.

**Review by r-ash — APPROVED** [review]
